src/components/validated_input.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <div>
+    <input id="input"
+           :class="{error : errors.length > 0}"
+           type="text"
+           :value="input_value"
+           @input="$e => _change_input($e.target.value)"/>
+    <h4 id="invalid-input-header"
+        v-if="!is_valid()"
+        class="error-text">Invalid input!</h4>
+    <ul>
+      <li v-for="error of errors"
```

Comment:
Should we add something to let users style the error messages? Maybe even a scoped slot?